src/features/pastelDB/__tests__/pastelDBThread.test.ts

```diff
@@ -295,179 +152,380 @@ describe('PastelDBThread', () => {
+      branchlen: 0,
+      hash: 'chain-tips-hash-1234578',
+      height: 0,
+      status: '',
+    })
+  })
+
+  test('fetchBlocksubsidy works', async () => {
+    // Arrange
+    const rpcSpy = jest
+      .spyOn(pastelRPC, 'rpc')
+      .mockResolvedValue(mockBlockSubSidy)
+    const insertBlocksubsidySpy = jest
+      .spyOn(pastelDBLib, 'insertBlocksubsidy')
+      .mockImplementation()
+
+    // Act
+    await pastelDBThread.fetchBlocksubsidy({
+      pastelDB: pastelDB.db,
+      rpcConfig: mockConfig,
+    })
+
+    // Assert
+    expect(rpcSpy).toHaveBeenCalled()
+    expect(rpcSpy).toHaveBeenCalledWith('getblocksubsidy', [], mockConfig)
+
+    expect(insertBlocksubsidySpy).toHaveBeenCalled()
+    expect(insertBlocksubsidySpy).toHaveBeenCalledWith(pastelDB.db, {
+      governance: 0,
+      masternode: 0,
+      miner: 456789,
+    })
+  })
+
+  test('fetchWalletInfo works', async () => {
+    // Arrange
+    const rpcSpy = jest
+      .spyOn(pastelRPC, 'rpc')
+      .mockResolvedValue(mockWalletInfo)
+    const insertWalletinfoSpy = jest
+      .spyOn(pastelDBLib, 'insertWalletinfo')
+      .mockImplementation()
+
+    // Act
+    await pastelDBThread.fetchWalletInfo({
+      pastelDB: pastelDB.db,
+      rpcConfig: mockConfig,
+    })
+
+    // Assert
+    expect(rpcSpy).toHaveBeenCalled()
+    expect(rpcSpy).toHaveBeenCalledWith('getwalletinfo', [], mockConfig)
+
+    expect(insertWalletinfoSpy).toHaveBeenCalled()
+    expect(insertWalletinfoSpy).toHaveBeenCalledWith(pastelDB.db, {
+      balance: 0,
+      immature_balance: 0,
+      keypoololdest: 0,
+      keypoolsize: 0,
+      paytxfee: 0,
+      seedfp: '',
+      txcount: 0,
+      unconfirmed_balance: 0,
+      walletversion: 456789,
+    })
+  })
+
+  test('fetchTotalbalance works', async () => {
+    // Arrange
+    const rpcSpy = jest
+      .spyOn(pastelRPC, 'rpc')
+      .mockResolvedValue(mockTotalBalance)
+    const insertTotalbalanceSpy = jest
+      .spyOn(pastelDBLib, 'insertTotalbalance')
+      .mockImplementation()
+
+    // Act
+    await pastelDBThread.fetchTotalbalance({
+      pastelDB: pastelDB.db,
+      rpcConfig: mockConfig,
+    })
+
+    // Assert
+    expect(rpcSpy).toHaveBeenCalled()
+    expect(rpcSpy).toHaveBeenCalledWith('z_gettotalbalance', [], mockConfig)
+
+    expect(insertTotalbalanceSpy).toHaveBeenCalled()
+    expect(insertTotalbalanceSpy).toHaveBeenCalledWith(pastelDB.db, {
+      private: '',
+      total: '',
+      transparent: 'totalbalance-transparent',
+    })
+  })
+
+  test('fetchListaddresses works', async () => {
+    // Arrange
+    const rpcSpy = jest
+      .spyOn(pastelRPC, 'rpc')
+      .mockResolvedValue(mockListAddresses)
+    const insertListaddressesSpy = jest
+      .spyOn(pastelDBLib, 'insertListaddresses')
+      .mockImplementation()
+
+    // Act
+    await pastelDBThread.fetchListaddresses({
+      pastelDB: pastelDB.db,
+      rpcConfig: mockConfig,
+    })
+
+    // Assert
+    expect(rpcSpy).toHaveBeenCalled()
```

Comment:
If there is `toHaveBeenCalledWith` in your test, `toHaveBeenCalled` is not required anymore